libs/util/lia-exporter/src/lib/question-utils.ts

```diff
@@ -0,0 +1,417 @@
+		if (i === acceptedAnswers.length - 1) {
+			script += `input == "${answer.value}"\n`;
+		} else {
+			script += `input == "${answer.value}" ||`;
+		}
+	}
+	script += "</script>\n";
+	return script;
+}
+
+/**
+ * Function that gets all answer blocks from a cloze style question.
+ * It does so by iterating over the text looking for the start of a block and then counting curly braces to make sure
+ * it finds the end while still allowing for any number of nested structures.
+ */
+const startOfAnswerBlock = RegExp(/\{\s*[TC]\s*:\s*.*?\s*/); // searches start of answer option block
+
+function getAllClozeAnswerBlocks(
+	text: string
+): { indexStartAnswerBlock: number; indexEndAnswerBlock: number; answerBlock: string }[] {
+	const matches: {
+		indexStartAnswerBlock: number;
+		indexEndAnswerBlock: number;
+		answerBlock: string;
+	}[] = [];
+	const chars = [...text];
+	let curlyBracketsCount = 0; //indicator/ counter for {
+	let indexStartAnswerBlock = 0; //the index where the { is found
+	let indexEndAnswerBlock = 0; //the index where the last closing } is found
+	let foundEndOfBlock = false; //used to indicate that a cloze answer is fully found and allows the conversion to start
+	let foundStartOfBlock = false;
+
+	for (let i = 0; i < chars.length; i++) {
+		if (chars[i] === "{") {
+			curlyBracketsCount++;
+			if (!foundStartOfBlock) {
+				foundStartOfBlock = startOfAnswerBlock.test(text.substring(i, i + 5)); //i+5 as this is our tolerance for correct syntax
+				if (foundStartOfBlock) {
+					indexStartAnswerBlock = i;
+					curlyBracketsCount = 1;
+				}
+			}
+		}
+
+		if (chars[i] === "}") {
+			curlyBracketsCount--;
+			if (curlyBracketsCount === 0 && foundStartOfBlock) {
+				foundEndOfBlock = true;
+				foundStartOfBlock = false;
+			}
+		}
+		if (foundEndOfBlock) {
+			indexEndAnswerBlock = i + 1;
+			matches.push({
+				indexStartAnswerBlock: indexStartAnswerBlock,
+				indexEndAnswerBlock: indexEndAnswerBlock,
+				answerBlock: text.substring(indexStartAnswerBlock, indexEndAnswerBlock)
+			});
+			foundEndOfBlock = false;
+		}
+	}
+	return matches;
+}
+const removeSelfLearAnswerStructure = /(?<!\\)([\]|[])/g; //Our cloze quizzes use [] to mark correct answers, we do not need this
+const removeClosingCurlyBracesOfAnswer = /}(?=[^}]*$)/; //Removes only the last curly braces of the answer and stops looking ahead, replaces it with ]]
+
+/**
+ * The method transforms the cloze answers for a given array. First it replaces the starting and ending curly braces with [[ and ]]
+ *
+ * @param param0 This is a triplet of the start index, the end index and the answer block itself. Only the text will be transformed.
+ * @param conversionError In case we encounter un convertible answer blocks we propagate the error
+ * @returns
+ */
+
+function transformClozeAnswerBlock(
+	{
+		indexStartAnswerBlock,
+		indexEndAnswerBlock,
+		answerBlock
+	}: { indexStartAnswerBlock: number; indexEndAnswerBlock: number; answerBlock: string },
```

Comment:
ich sehe das häufiger. ein named type könnte für das triplet hilfreich sein.

Reply:
Hybe einen named Type eingeführt, wurde aber nur in 2 Funktionen verwendet: 62ba4391